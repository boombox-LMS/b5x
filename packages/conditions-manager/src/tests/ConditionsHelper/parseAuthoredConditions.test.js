const { ConditionsHelper } = require("../../../dist/index");
const _ = require("lodash");

describe("ConditionsHelper.parseAuthoredConditions is working as expected", () => {
  test("parseAuthoredConditions returns an empty array when given a string with no content", () => {
    const rawConditions = "           ";
    const expectedResult = [];
    const result = ConditionsHelper.parseAuthoredConditions(rawConditions);
    expect(_.isEqual(result, expectedResult)).toBe(true);
  });

  test("parseAuthoredConditions returns a one-item conditions list when given a single condition", () => {
    const rawConditions = "reader-mood is good";
    const expectedResult = [
      {
        type: "condition",
        data: { alias: "reader-mood", operator: "is", targetValue: "good" },
      },
    ];
    const result = ConditionsHelper.parseAuthoredConditions(rawConditions);
    expect(_.isEqual(result, expectedResult)).toBe(true);
  });

  test("complex parseAuthoredConditions results match snapshots", () => {
    const inputs = [
      "reader-mood is good and favorite-color is blue and location is Seattle",
      "(reader-mood is good and favorite-color is blue) or location is Seattle",
      "(reader-mood is good) or ((favorite-color is blue) and (location is Seattle))",
    ];

    inputs.forEach((input) => {
      const result = ConditionsHelper.parseAuthoredConditions(input);
      expect(result).toMatchSnapshot();
    });
  });
});
