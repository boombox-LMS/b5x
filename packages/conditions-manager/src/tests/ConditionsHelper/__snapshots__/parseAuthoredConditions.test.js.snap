// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionsHelper.parseAuthoredConditions is working as expected complex parseAuthoredConditions results match snapshots 1`] = `
[
  [
    {
      "data": {
        "alias": "reader-mood",
        "operator": "is",
        "targetValue": "good",
      },
      "type": "condition",
    },
  ],
  {
    "data": "and",
    "type": "conjunction",
  },
  [
    {
      "data": {
        "alias": "favorite-color",
        "operator": "is",
        "targetValue": "blue",
      },
      "type": "condition",
    },
  ],
  {
    "data": "and",
    "type": "conjunction",
  },
  [
    {
      "data": {
        "alias": "location",
        "operator": "is",
        "targetValue": "Seattle",
      },
      "type": "condition",
    },
  ],
]
`;

exports[`ConditionsHelper.parseAuthoredConditions is working as expected complex parseAuthoredConditions results match snapshots 2`] = `
[
  [
    [
      {
        "data": {
          "alias": "reader-mood",
          "operator": "is",
          "targetValue": "good",
        },
        "type": "condition",
      },
    ],
    {
      "data": "and",
      "type": "conjunction",
    },
    [
      {
        "data": {
          "alias": "favorite-color",
          "operator": "is",
          "targetValue": "blue",
        },
        "type": "condition",
      },
    ],
  ],
  {
    "data": "or",
    "type": "conjunction",
  },
  [
    {
      "data": {
        "alias": "location",
        "operator": "is",
        "targetValue": "Seattle",
      },
      "type": "condition",
    },
  ],
]
`;

exports[`ConditionsHelper.parseAuthoredConditions is working as expected complex parseAuthoredConditions results match snapshots 3`] = `
[
  [
    {
      "data": {
        "alias": "reader-mood",
        "operator": "is",
        "targetValue": "good",
      },
      "type": "condition",
    },
  ],
  {
    "data": "or",
    "type": "conjunction",
  },
  [
    [
      {
        "data": {
          "alias": "favorite-color",
          "operator": "is",
          "targetValue": "blue",
        },
        "type": "condition",
      },
    ],
    {
      "data": "and",
      "type": "conjunction",
    },
    [
      {
        "data": {
          "alias": "location",
          "operator": "is",
          "targetValue": "Seattle",
        },
        "type": "condition",
      },
    ],
  ],
]
`;
